storage/filestore.go

```diff
@@ -221,7 +221,7 @@ func (f *FilesystemStore) Set(name string, meta []byte) error {
 	os.RemoveAll(fp)
 
 	// Write the file to disk
-	if err = ioutil.WriteFile(fp, meta, f.perms); err != nil {
+	if err = ioutil.WriteFile(fp, meta, f.perms&notary.PermExecutableMask); err != nil {
```

Comment:
The function naming is poor, but the intent was to have utility initializers to create public, `NewSimpleFileStore`, and private, `NewFilesystemStore`, file storage, with the ability to always set more explicit options via function they both call: `NewFileStore`.

It would be fine to create more wrappers if necessary, but I think _everything_ is using `NewFilesystemStore` at the moment (#1010 removes `NewSimpleFileStore`), so we should just modify the permissions for everyone there.

We're also thinking of adding https://gas.css.bio/#docker/notary as a lint check so we might want to cut the permissions down to `0700` / `0600` and have no concept of public files.